src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +39,25 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={handlerQuery}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {query && (
+            <>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="clearSearchButton"
+                type="button"
+                className="delete"
+                onClick={handlerQueryCancel}
+              />
+            </>
```

Comment:
Remove redundant react.fragment
```suggestion
              <button
                data-cy="clearSearchButton"
                type="button"
                className="delete"
                onClick={handlerQueryCancel}
              />
```